### PR TITLE
Don't fail design time builds if the compiler hasn't been restored

### LINKED ISSUE #1890: Opening a document from a project with unresolved nuget references opens it in Misc files project
1. clone any github project that has nuget references, but do not resolve nuget references just yet.
2. open a document from this project.

**Actual**:
Document opens as a Misc files project. No errors in error list.

**Expected**: 
Document opens in the correct project context and squiggles show up for members that have missing references.

cc @dpoeschl

### PR DESCRIPTION
**Pull request by Pilchie**
Fixes #1890

### CODE REVIEW

**Comment by Pilchie** [comment]
@jasonmalinowski @agocke

File: build/Targets/VSL.Settings.targets [hunk 1]
```diff
@@ -102,7 +102,7 @@
             Condition="Exists('$(RoslynDiagnosticsPropsFilePath)') And ('$(UseRoslynAnalyzers)' == 'true')" />
   </ImportGroup>
 
-  <Target Name="EnsureGlobalNuGets" Condition="'$(BootstrapBuildPath)' == ''">
+  <Target Name="EnsureGlobalNuGets" Condition="'$(BootstrapBuildPath)' == '' AND '$(BuildingProject)' == 'true'">
```
Comment (jasonmalinowski) [inline build/Targets/VSL.Settings.targets]: Should this be '$(BuildingProject)' != 'false'? I don't know if or when we care about points when it's unset.

Comment (Pilchie) [inline build/Targets/VSL.Settings.targets]: Yeah, good point.

**Comment by agocke** [comment]
The toolset compiler could contain a new version of the build task, which would effect even design-time builds, would it not?

**Comment by jasonmalinowski** [comment]
It could, but the philosophy is we don't care for the sake of the design time build. It's better off running with the standard task (which currently has no real harm) than completely breaking the design time build and leaving the user with a broken IDE.

**Comment by agocke** [comment]
:+1: Then :)

**Comment by Pilchie** [comment]
The EnsureGlobalNugets target was removed by 64e32e153ee4b0e15463fbb61f21cc51fffd62ea, so this is no longer the fix here :frowning:

**Pull request closed without merging**